cmd/kubeadm/app/util/version.go

```diff
@@ -72,3 +72,12 @@ func KubernetesReleaseVersion(version string) (string, error) {
 	}
 	return "", fmt.Errorf("version %q doesn't match patterns for neither semantic version nor labels (stable, latest, ...)", version)
 }
+
+// KubernetesVersionToImageTag is helper function that replaces all
+// non-allowed symbols in tag strings with underscores.
+// Image tag can only contain lowercase and uppercase letters, digits,
+// underscores, periods and dashes.
+func KubernetesVersionToImageTag(version string) string {
```

Comment:
Keep in mind that this version already matches `kubeReleaseRegex` above. So the only disallowed char allowed in semver but not in an image tag is `+` AFAIK.
Therefore, could you make this clearer that the case here is about replacing `+` with `_` for the CI case where images are actually pushed with `_` instead of `+` instead of seemingly randomly converting strange chars to underscores.